### DISCUSSION TITLE
How can I generate prisma schema from DB with mapping different name styles

### DISCUSSION THREAD

**Discussion by Gnosnay** [prisma/prisma#18263] [Q&A] [answered]
hi guys thank you so much for your reading!

I met one issue when generating one prisma schema from real DB table.

the DB schema is like:
```
user_tab
id, create_at, username
```

when i run `npx prisma db pull`,  i get the snake case naming like:
```prisma
model user_tab {
  id                 Int               @id @default(autoincrement())
  username               String          @db.VarChar(64)
  created_at    DateTime  @db.DateTime(6)
}
```

how can i give some options to prisma to help me mapping different naming styles? what i want is like:
```prisma
model User {
  id                 Int               @id @default(autoincrement())
  username               String          @db.VarChar(64)
  createdAt    DateTime  @db.DateTime(6)
}
```

**Comment by nurul3101** [comment, marked as answer]
Hello @Yansongsongsong 👋 

It looks like you are looking for [@map](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#map) attribute. Can you check it out?

**Reply by Gnosnay** [reply]
yes i knew it.

but it can only be used when `schema.prisma` was generated. my question is during the generation of `schema.prisma`, how can i configure to let it mapping the different table and column name

**Reply by Gnosnay** [reply]
still thank you so much for your reply

**Reply by nurul3101** [reply]
Oh, I get it now!
We have a related feature request here:
- https://github.com/prisma/prisma/issues/1184

which intends to support the same use case that you have. If you can add a 👍 to the feature request, then it would help our product team in prioritising it. Also, I would recommend you to check out `prisma-case-format` as mentioned in this [comment](https://github.com/prisma/prisma/issues/1184#issuecomment-1411828318).

**Reply by Gnosnay** [reply]
nice that is what i look for! thank you so much

**Comment by Gnosnay** [comment]
work around: https://github.com/iiian/prisma-case-format

**Comment by vpmedia** [comment]
Since prisma-case-format is not maintained anymore, I've created an updated fork:
https://github.com/vpmedia/prisma-schema-remap
